### PR TITLE
Cleanup DebugInfo and Value

### PR DESCRIPTION
**Pull request by gramalingam**
Cleanup.

* Rename DebugInfo to SourceInfo, and some refactoring
* Refactor calls/use of DebugInfo
* Rename Value class to SymbolValue and minor refactoring

### CODE REVIEW

File: onnxscript/analysis.py [hunk 1]
```diff
@@ -145,24 +143,20 @@ def visitBlock(block, live_out):
         if isinstance(stmt, ast.FunctionDef):
             return live_out
         try:
-            if stmt.value.func.id == "print":
+            if stmt.value.func.id == "print":  # type: ignore[attr-defined]
                 # Any call to print function are ignored.
                 return live_out
         except (TypeError, AttributeError):
```
Comment (justinchuby) [inline onnxscript/analysis.py:149]: nit: from CodeQL: 'except' clause does nothing but pass and there is no explanatory comment.

Helpful to add a comment?

File: onnxscript/converter.py [hunk 2]
```diff
@@ -913,13 +904,11 @@ def translate_stmt(self, node, index_of_stmt=None):
                 return None
         except (TypeError, AttributeError):
```
Comment (justinchuby) [inline onnxscript/converter.py:905]: CodeQL

File: onnxscript/test/analysis_test.py [hunk 3]
```diff
@@ -96,8 +97,8 @@ def while_eg(x):
 
 class TestExposedUses(unittest.TestCase):
     def assertUses(self, f, expected):
-        ast = main.get_ast(f)  # pylint: disable=redefined-outer-name
-        result = analysis.exposed_uses(ast.body, converter.Converter())
+        source, ast = main.get_src_and_ast(f)  # pylint: disable=redefined-outer-name
```
Comment (justinchuby) [inline onnxscript/test/analysis_test.py:100]: Since this line is touched - possible to rename `ast` to avoid `disable=redefined-outer-name`?

**Review by justinchuby — APPROVED** [review]

**Comment by gramalingam** [comment]
@justinchuby : any idea about the lintrunner CI failure? Pylint is not recognizing the `-> NoReturn` annotation, even though pylint documentation seems to imply it should. Further, `pylint: disable=R1710` doesn't seem to work either ...

**Comment by justinchuby** [comment]
Oh that's right. It's no return. I would add this line to the top of the file

```
# pylint: disable=inconsistent-return-statements
```

**Comment by gramalingam** [comment]
> Oh that's right. It's no return. I would add this line to the top of the file
> 
> ```
> # pylint: disable=inconsistent-return-statements
> ```
> 
> If we want more fine controls, I would add `# pylint: disable=inconsistent-return-statements` and `# pylint: enable=inconsistent-return-statements` to before and after the class

Doesn't it work at function/method scope level? That's what I tried.

**Comment by justinchuby** [comment]
Actually no

```
def translate_callee_expr(self, node) -> values.Op:  # pylint: disable=R1710
```

doesn't look very long

File: onnxscript/test/analysis_test.py (hunk 3 above)
Comment (gramalingam) [reply onnxscript/test/analysis_test.py]: Done

File: onnxscript/sourceinfo.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
```
[the remark this answers is not in the document]
Comment (gramalingam) [reply onnxscript/sourceinfo.py]: Done

File: onnxscript/converter.py (hunk 2 above)
Comment (gramalingam) [reply onnxscript/converter.py]: Done

File: onnxscript/analysis.py (hunk 1 above)
Comment (gramalingam) [reply onnxscript/analysis.py]: Cleaned up, thanks!

**Comment by justinchuby** [comment]
LGTM!

**Pull request merged**